hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieTimeline.java

```diff
@@ -246,6 +257,19 @@ public interface HoodieTimeline extends Serializable {
    */
   HoodieTimeline findInstantsInClosedRange(String startTs, String endTs);
 
+  /**
+   * Create a new Timeline with instants after or equals startTs and before or on endTs
+   * by completionTime.
+   */
+  HoodieTimeline findInstantsInClosedRangeByCompletionTime(String startTs, String endTs);
+
```

Comment:
Looks like it is relevent with this change: https://github.com/apache/hudi/pull/10255

Reply:
Oh yea, they're relevant, by the way, this pr tries to read partial files in one commit, not sure we can integrate it in https://github.com/apache/hudi/pull/10255

Reply:
> this pr tries to read partial files in one commit,

Not sure we should, the basic granuarity for data compleness is a whole `commit`

Reply:
Rate limit on the `commit` level means on the `write operation` level(one write operation equals to one commit), However, this approach may not be sufficient when we need to handle incoming data that is divided based on each write operation. Let me provide two scenarios to illustrate this:

1. Batch write -> HUDI -> Streaming read, batch job could be with more resources than the streaming job, there could be `insert` some data, `insert_overwrite` across many partitions. The streaming job could have highly risk to fail if we can only limit the data at commit level. We have to adjust large resources for this long running streaming job to avoid it.
2. streaming write -> HUDI -> different downstream streaming read jobs. Each downstream job might have different resource limits, some jobs could have efficient resources allowing them to have large commits, but some's are not(they can tolerate slightly delayed data freshness), if we only keep at commit level, all downstream jobs must keep at least the same resource with the upstream streaming job.

Is there any issue with reading partial files within a single commit? I'm considering this as a potential improvement that would offer users more choices